src/settingsdialogplugins.cpp

```diff
@@ -150,6 +194,97 @@ void PluginsSettingsTab::closing()
   storeSettings(ui->pluginsList->currentItem());
 }
 
+void PluginsSettingsTab::on_checkboxEnabled_clicked(bool checked)
+{
+  // Retrieve the plugin:
+  auto *item = ui->pluginsList->currentItem();
+  if (!item || !item->data(0, ROLE_PLUGIN).isValid()) {
+    return;
+  }
+  IPlugin* plugin = this->plugin(item);
+  const auto& requirements = m_pluginContainer->requirements(plugin);
+
+  // User wants to enable:
+  if (checked) {
+    auto problems = requirements.problems();
+    if (!problems.empty()) {
+      QStringList descriptions;
+      for (auto& problem : problems) {
+        descriptions.append(problem.shortDescription());
+      }
+      QMessageBox::warning(
+        parentWidget(), QObject::tr("Cannot enable plugin"),
+        QObject::tr("<p>This plugin cannot be enabled:<p><ul>%1</ul>")
+        .arg("<li>" + descriptions.join("</li><li>") + "</li>"), QMessageBox::Ok);
+      ui->enabledCheckbox->setChecked(false);
+      return;
+    }
+
+    m_pluginContainer->setEnabled(plugin, true, false);
+  }
+  else {
+    // Custom check for proxy + current game:
+    if (m_pluginContainer->implementInterface<IPluginProxy>(plugin)) {
+
+      // Current game:
+      auto* game = m_pluginContainer->managedGame();
+      if (m_pluginContainer->requirements(game).proxy() == plugin) {
+        QMessageBox::warning(
+          parentWidget(), QObject::tr("Cannot disable plugin"),
+          QObject::tr("This plugin is used by the current game plugin and cannot disabled."), QMessageBox::Ok);
+          ui->enabledCheckbox->setChecked(true);
+          return;
+      }
+
+      // Check the proxied plugins:
+      auto proxied = requirements.proxied();
+      QStringList pluginNames;
+      for (auto& p : proxied) {
+        pluginNames.append(p->localizedName());
+      }
+      pluginNames.sort();
+      QString message = QObject::tr(
+        "<p>Disabling this plugin will prevent the following plugins from working:</p><ul>%1</ul>"
+        "<p>Do you want to continue? You will need to restart ModOrganizer2 for the change to take effect.</p>")
+        .arg("<li>" + pluginNames.join("</li><li>") + "</li>");
+      if (QMessageBox::critical(
+        parentWidget(), QObject::tr("Really disable plugin?"), message,
+        QMessageBox::Yes | QMessageBox::No) == QMessageBox::No) {
+        ui->enabledCheckbox->setChecked(true);
+        return;
+      }
+
+    }
+
+    // Check if the plugins is required for other plugins:
+    auto requiredFor = requirements.requiredFor();
```

Comment:
Do we have any plugin right now that would trigger this?

Reply:
I think the fnis plugins should, but there's no PR.

Reply:
I don't think there are any plugins actually. The FNIS tool has a master dependency which won't trigger this (you can only enable or disable all 3 plugins at once).